--- src/map/battle.cpp
+++ src/map/battle.cpp
@@ -123,6 +123,13 @@
 void battle_damage(struct block_list* src, struct block_list* target, int64 damage, int64 div, uint16 skill_id, uint16 skill_lv, enum e_battle_dmg_type type)
 {
 	struct status_data *sstatus = status_get_status_data(src);
+	struct map_session_data *sd = nullptr;
+
+	// WOE Stats: Track player damage in GvG maps
+	if (target != nullptr && target->type == BL_PC && map_flag_gvg(target->m)) {
+		sd = map_id2sd(target->id);
+		pc_adddamage(sd, damage);
+	}
 
 	nullpo_retv(src);
 	nullpo_retv(target);
