--- src/map/itemdb.h.orig
+++ src/map/itemdb.h
@@ -123,6 +123,12 @@ struct item_data {
 	void (*unequip_script) (struct map_session_data *sd);
 };
 
+// Random Options System
+void itemdb_random_option_init(void);
+void itemdb_roll_random_options(struct item *item, int type);
+int itemdb_check_random_options(int nameid);
+
 // Item Trade Restrictions
 void itemdb_restraint_update(bool silent); 